sanic/headers.py

```diff
@@ -24,14 +24,132 @@ def parse_content_header(value: str) -> typing.Tuple[str, Options]:
+    :return: dict with matching keys (lowercase) and values, or None.
+    """
+    header = headers.getall("forwarded", None)
+    secret = config.FORWARDED_SECRET
+    if header is None or not secret:
+        return None
+    header = ",".join(header)  # Join multiple header lines
+    if secret not in header:
+        return None
+    # Loop over <separator><key>=<value> elements from right to left
+    ret = sep = pos = None
+    found = False
+    for m in _rparam.finditer(header[::-1]):
+        # Start of new element? (on parser skips and non-semicolon right sep)
+        if m.start() != pos or sep != ";":
+            if found:
+                return normalize(ret)
+            ret = {}
+        pos = m.end()
+        val_token, val_quoted, key, sep = m.groups()
+        key = key.lower()[::-1]
+        val = (val_token or val_quoted.replace('"\\', '"'))[::-1]
+        ret[key] = val
+        if key == "secret" and val == secret:
+            found = True
+        if found and sep != ";":
+            return normalize(ret)
+    return normalize(ret) if found else None
+
+
+def parse_xforwarded(headers, config):
+    """Parse X-Real-IP, X-Scheme and X-Forwarded-* headers."""
+    proxies_count = config.PROXIES_COUNT
+    if not proxies_count:
+        return None
+    h1, h2 = config.REAL_IP_HEADER, config.FORWARDED_FOR_HEADER
+    addr = h1 and headers.get(h1)
+    forwarded_for = h2 and headers.getall(h2, None)
+    if not addr and forwarded_for:
+        assert proxies_count == -1 or proxies_count > 0, config.PROXIES_COUNT
+        # Combine, split and filter multiple headers' entries
+        proxies = (p.strip() for h in forwarded_for for p in h.split(","))
+        proxies = [p for p in proxies if p]
+        try:
+            addr = proxies[-proxies_count] if proxies_count > 0 else proxies[0]
+        except IndexError:
+            return None
+    if not addr:
+        return None
+    other = (
+        (key, headers.get(header))
+        for key, header in (
+            ("proto", "x-scheme"),
+            ("proto", "x-forwarded-proto"),  # Overrides X-Scheme if present
+            ("host", "x-forwarded-host"),
+            ("port", "x-forwarded-port"),
+            ("path", "x-forwarded-path"),
+        )
+    )
+    return normalize({"for": addr, **{k: v for k, v in other if v}})
+
+
+_ipv6 = "(?:[0-9A-Fa-f]{0,4}:){2,7}[0-9A-Fa-f]{0,4}"
+_ipv6_re = re.compile(_ipv6)
+_host_re = re.compile(
+    r"((?:\[" + _ipv6 + r"\])|[a-zA-Z0-9.\-]{1,253})(?::(\d{1,5}))?"
+)
+
+
+def parse_host(host):
+    m = _host_re.fullmatch(host)
+    if not m:
+        return None, None
+    host, port = m.groups()
+    return host.lower(), port and int(port)
+
+
+def bracketv6(addr):
+    return f"[{addr}]" if _ipv6_re.fullmatch(addr) else addr
+
+
+def normalize(fwd: dict) -> dict:
+    """Normalize and convert values extracted from forwarded headers.
+    Modifies fwd in place and returns the same object.
+    """
+    if "proto" in fwd:
```

Comment:
```python
def handle_host(forward):
    host, port = parse_host(forward["host"])
    if host:
        forward["host"] = host
        if port:
            forward["port"] = port
    else:
        del forward["host"]

actions = {
    "proto": {
        "handler": None,
    },
    "by": {
        "handler": bracketv6
    },
    "for": {
        "handler": bracketv6
    },
    "port": {
        "handler": handle_port
    },
    "host": {
        "handler": handle_host
    }
}

for key in fwd:
    try:
        actions[key]["handler"](fwd)
    except:
        fwd.pop("port", None)
```
Can you please check if it make sense to have this section defined this way without all the `if` checks?

Reply:
There is enough variation in the tests to make this approach somewhat unfeasible. If there wasn't that host split into host and port, it might just do.

Reply:
I don't really have a problem with the current implementation. Just that I am more for lookup over if checks that's all. If you think it won't help in speading up things, then what you already have is perfectly fine.